### PR TITLE
ESASky2.0, Windows support, bug fixes, and more tests

### PR DESCRIPTION
**Pull request by imbasimba**

### CODE REVIEW

**Comment by imbasimba** [comment]
Issue #980 is addressed among other things

File: astroquery/esasky/core.py [hunk 1]
```diff
@@ -585,15 +586,19 @@ def _remove_extra_herschel_directory(self, file_and_directory_name,
                                          directory_path):
         full_directory_path = os.path.abspath(directory_path)
         file_name = file_and_directory_name[file_and_directory_name.index("/") + 1:]
+        
+        print("full path: %s" %(full_directory_path))
```
Comment (bsipocz) [inline astroquery/esasky/core.py]: could you use the logger for any prints (e.g these seems to be ``log.info``), or introduce a verbose variable?

File: astroquery/esasky/core.py [hunk 2]
```diff
@@ -514,8 +513,11 @@ def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
             for index in range(len(maps_table)):
                 product_url = (maps_table[self.__PRODUCT_URL_STRING][index]
                                .decode('utf-8'))
-                observation_id = (maps_table[self.__OBSERVATION_ID_STRING][index]
-                                  .decode('utf-8'))
+                if(mission.lower() == self.__HERSCHEL_STRING):
```
Comment (bsipocz) [inline astroquery/esasky/core.py]: no need for the parens here and in the next line

**Comment by imbasimba** [comment]
@bsipocz I think this is ready for merging. The Travis problem seems unrelated

**Comment by bsipocz** [comment]
Thank you @imbasimba. Indeed that failure was unrelated and has been dealt with upstream.

**Pull request merged**